frontend/src/api/sessions.ts

```diff
@@ -1,17 +1,39 @@
 import {API_URL} from "../config";
+import {TPlayer} from "./players";
 
-export type Session = {
+export type TSessionPlayers = {
+    sessionCode: string,
+    playerCode: string,
+    player?: TPlayer,
+    session?: TSession //todo this property is not returning anything from the api
+}
+
+export type TSession = {
     code: string,
     description: string,
     minGamesRequired: number,
     isActive: boolean,
-    sessionPlayers: Array<any> //todo sessionPlayer
+    sessionPlayers: Array<TSessionPlayers>
 }
 
 export const sessionApi = {
-    getAll: async (): Promise<Session[]> => await (await fetch(API_URL + "/sessions")).json(),
-    getOne: async (code: string): Promise<Session> => await (await fetch(`${API_URL}/sessions/${code}`)).json(),
-    create: async (body: Partial<Session>): Promise<Response> => await fetch(`${API_URL}/sessions`, {
+    getAll: async (): Promise<TSession[]> => await (await fetch(API_URL + "/sessions")).json(),
+    getOne: async (code: string): Promise<TSession> => await (await fetch(`${API_URL}/sessions/${code}`)).json(),
+    addPlayers: async (players: Array<TSessionPlayers>): Promise<Response> => await fetch(`${API_URL}/sessions/players`, {
+        method: "PUT",
+        body: JSON.stringify(players),
+        headers: {
+            'Accept': 'application/json',
+            'Content-Type': 'application/json'
+    }}),
+    deletePlayers: async (players: Array<TSessionPlayers>): Promise<Response> => await fetch(`${API_URL}/sessions/players`, {
+        method: "DELETE",
+        body: JSON.stringify(players),
+        headers: {
+            'Accept': 'application/json',
+            'Content-Type': 'application/json'
+        }}),
+    create: async (body: Partial<TSession>): Promise<Response> => await fetch(`${API_URL}/sessions`, {
         method: "PUT",
         body: JSON.stringify(body),
         headers: {
```

Comment:
need to reduce boilerplate code

Reply:
I will refactor later on, I will try to create an abstraction of the http client to encapsulate error handling, etc